templates/default/hadoop-init.erb

```diff
@@ -0,0 +1,155 @@
+#! /bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Starts a <%= @options['desc'] %>
+#
+# chkconfig: 345 85 15
+# description: Starts a <%= @options['desc'] %>
+# processname: <%= @options['process'] %>
+#
+### BEGIN INIT INFO
+# Provides:          <%= @options['name'] %>
+# Required-Start:    $syslog $remote_fs
+# Should-Start:
+# Required-Stop:     $syslog $remote_fs
+# Should-Stop:
+# Default-Start:     3 4 5
+# Default-Stop:      0 1 2 6
+# Short-Description: Starts a <%= @options['desc'] %>
+### END INIT INFO
+
+. /lib/lsb/init-functions
+[ -f /etc/default/hadoop ] && . /etc/default/hadoop
+
+# Autodetect JAVA_HOME if not defined
+if [ -e /usr/libexec/bigtop-detect-javahome ]; then
+  . /usr/libexec/bigtop-detect-javahome
+elif [ -e /usr/lib/bigtop-utils/bigtop-detect-javahome ]; then
+  . /usr/lib/bigtop-utils/bigtop-detect-javahome
+fi
+[ -f /etc/profile.d/jdk.sh ] && . /etc/profile.d/jdk.sh
+
+RETVAL_SUCCESS=0
+
+STATUS_RUNNING=0
+STATUS_DEAD=1
+STATUS_DEAD_AND_LOCK=2
+STATUS_NOT_RUNNING=3
+STATUS_DEBIAN_NOT_RUNNING=4
```

Comment:
nitpick: probably should rename to ``STATUS_PROGRAM_NOT_RUNNING`` or similar to reflect this is not debian-specific